### PR TITLE
feat(Benchmark): RHICOMPL-1538 searching by supported_os_minor_versions

### PR DESCRIPTION
**Pull request by vkrizan**
Adds scope `supported_os_minor_versions` for `Benchmark` model.

Adds scoped search fitlering `supported_os_minor_version`.
Examples:
```
supported_os_minor_version = 9
supported_os_minor_version ^ (9, 10)
os_major_version = 7 and supported_os_minor_version ^ (8, 9)
```

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: app/models/xccdf/benchmark_searching.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
```
Comment (skateman) [inline app/models/xccdf/benchmark_searching.rb:1]: I think the right location for this file should be:
```
app/models/concerns/xccdf/benchmark_searching.rb
```
Or maybe:
```
app/models/xccdf/concerns/benchmark_searching.rb
```
But I think the first one is the more rails-y

Comment (vkrizan) [reply app/models/xccdf/benchmark_searching.rb:1]: The latter did not worked. The former can do it maybe...

Comment (akofink) [reply app/models/xccdf/benchmark_searching.rb:1]: I'm ok to have this outside of the Xccdf module. One day, we'll probably remove benchmark.rb from Xccdf as well. It was intended to show our models that have a 1 to 1 counterpart in the SCAP model, but we never actually followed up with it for rules, profiles, test results, etc., and I don't really see why it's useful anyway. To use it within benchmark.rb, all we'd need is the root namespace, `include ::BenchmarkSearching`.

File: app/models/xccdf/benchmark_searching.rb [hunk 2]
```diff
@@ -20,6 +20,12 @@ module BenchmarkSearching
         where(os_major_version_query(major, equals))
       }
 
+      scope :supported_os_minor_versions, lambda { |minor_versions|
+        supported_os_minor_clauses(minor_versions).reduce(none) do |ors, clause|
+          ors.or(where(*clause))
+        end
+      }
```
Comment (akofink) [inline app/models/xccdf/benchmark_searching.rb:31]: Seems to work as expected :+1: 

```rb
[8] pry(main)> Xccdf::Benchmark.os_major_version(7).supported_os_minor_versions([7, 8, 9]).count
   (0.5ms)  SELECT COUNT(*) FROM "benchmarks" WHERE "benchmarks"."ref_id" ILIKE '%RHEL-7' AND ((benchmarks.ref_id LIKE '%RHEL-6' AND benchmarks.version IN ('0.1.21','0.1.28','0.1.28')) OR (benchmarks.ref_id LIKE '%RHEL-7' AND benchmarks.version IN ('0.1.43','0.1.46','0.1.52')))
```

File: app/models/xccdf/benchmark_searching.rb [hunk 3]
```diff
@@ -48,6 +54,22 @@ def os_major_version_search(_filter, operator, value)
           parameter: [os_major_version_like_condition(value)] }
       end
 
+      def supported_os_minor_clauses(minor_versions)
+        minor_versions = [minor_versions].flatten.map(&:to_s)
+
+        SupportedSsg.latest_map.map do |major, major_ssgs|
+          ssg_versions = minor_versions.map do |minor_version|
+            major_ssgs[minor_version]&.version
+          end.compact
```
Comment (akofink) [inline app/models/xccdf/benchmark_searching.rb:67]: Nit: we're sending non-unique minor versions in some cases. It doesn't affect the query result, but we might want to add a `.uniq` here.

```rb
[8] pry(main)> Xccdf::Benchmark.os_major_version(7).supported_os_minor_versions([7, 8, 9]).count
   (0.5ms)  SELECT COUNT(*) FROM "benchmarks" WHERE "benchmarks"."ref_id" ILIKE '%RHEL-7' AND ((benchmarks.ref_id LIKE '%RHEL-6' AND benchmarks.version IN ('0.1.21','0.1.28','0.1.28')) OR (benchmarks.ref_id LIKE '%RHEL-7' AND benchmarks.version IN ('0.1.43','0.1.46','0.1.52')))
```

The `benchmarks.version IN ('0.1.21','0.1.28','0.1.28')` part.

**Review by akofink — COMMENTED** [review]
Seems to work great! Nice work getting this to just a single SQL query. The only issue is the included benchmark_searching concern, which we should be able to fix by just not having it in the Xccdf module. Thanks, @vkrizan!

File: app/models/xccdf/benchmark_searching.rb (hunk 3 above)
Comment (vkrizan) [reply app/models/xccdf/benchmark_searching.rb]: :+1:

File: app/models/xccdf/benchmark_searching.rb (hunk 1 above)
Comment (vkrizan) [reply app/models/xccdf/benchmark_searching.rb]: Moved to top model-level concern.

**Review by akofink — APPROVED** [review]
Thanks, @vkrizan!

**Pull request merged**